pkg/p2p/libp2p/connections_test.go

```diff
@@ -167,20 +168,26 @@ func TestStreamsMaxIncomingLimit(t *testing.T) {
 	}})
 	s2, overlay2 := newService(t, 1, libp2pServiceOpts{})
 
+	var smu sync.Mutex
```

Comment:
Maybe name it `streamsMu` and place it near `streams` to make it clearer what the mutex is guarding.